pvlib/data/gold/sdm.py

```diff
@@ -0,0 +1,832 @@
+                            "r_sh": iv_curve["r_sh"],
+                            "nNsVth": iv_curve["nNsVth"],
+                            "v_gold": iv_curve["v_gold"].tolist(),
+                            "i_gold": iv_curve["i_gold"].tolist(),
+                            "v_test": v_test.tolist(),
+                            "residual": v_test_terminal_res.tolist(),
+                            "max_abs_residual":
+                                v_test_terminal_max_abs_res.tolist(),
+                    }
+
+        # Compute summary performance statistics for this device
+        if "i_from_v" in test_func:
+            device_dict["i_from_v"]["performance"]["time_s"]["min"] = \
+                min(device_dict["i_from_v"]["performance"]["time_s"]["data"])
+            device_dict["i_from_v"]["performance"]["time_s"]["med"] = \
+                numpy.median(device_dict["i_from_v"]
+                             ["performance"]["time_s"]["data"])
+            device_dict["i_from_v"]["performance"]["time_s"]["max"] = \
+                max(device_dict["i_from_v"]["performance"]["time_s"]["data"])
+            device_dict["i_from_v"]["performance"]["time_s"]["avg"] = \
+                numpy.average(device_dict["i_from_v"]
+                              ["performance"]["time_s"]["data"])
+            device_dict["i_from_v"]["performance"]["time_s"]["std"] = \
+                numpy.std(device_dict["i_from_v"]
+                          ["performance"]["time_s"]["data"], ddof=1)
+
+        if "v_from_i" in test_func:
+            device_dict["v_from_i"]["performance"]["time_s"]["min"] = \
+                min(device_dict["v_from_i"]["performance"]["time_s"]["data"])
+            device_dict["v_from_i"]["performance"]["time_s"]["med"] = \
+                numpy.median(device_dict["v_from_i"]
+                             ["performance"]["time_s"]["data"])
+            device_dict["v_from_i"]["performance"]["time_s"]["max"] = \
+                max(device_dict["v_from_i"]["performance"]["time_s"]["data"])
+            device_dict["v_from_i"]["performance"]["time_s"]["avg"] = \
+                numpy.average(device_dict["v_from_i"]
+                              ["performance"]["time_s"]["data"])
+            device_dict["v_from_i"]["performance"]["time_s"]["std"] = \
+                numpy.std(device_dict["v_from_i"]
+                          ["performance"]["time_s"]["data"], ddof=1)
+
+        # Append the worst results and the benchmarking for this device to the
+        #  output list of devices
+        # For convienent post processing and analysis, floats are NOT in hex
+        #  format
+        results.append(device_dict)
+
+    # Unit test consumers should be able to pass/fail based on the worst
+    #  results returned for each device
+    return results
+
+
+# Command line option
+if __name__ == '__main__':
+
+    gold_dataset = make_gold_dataset()
+
+    # File locations are robust to running script from different directories
```

Comment:
This makes sense for a local, development installation, but I'm not sure what the right behavior should be for a standard installation. Perhaps the default should be the current directory? ``argparse`` is another option.

Reply:
I now use `argparse` to make this more flexible (user can provide the directory and/or the extension-less filename), but I still have the default behavior create/overwrite files in the `pvlib/data/gold` directory.